File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashMap={}
        if len(s)!=len(t):
            return False
        for i in t:
            if i in hashMap:
                hashMap[i]=hashMap[i]+1
            else:
                hashMap[i]=1
        for i in s:
            if i in hashMap:
                hashMap[i]=hashMap[i]-1
            else:
                return False
        for i in hashMap:
            if hashMap[i]!=0:
                return False
        return True


        

File: 242-valid-anagram/valid-anagram.java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {

        HashMap<Character, Integer> hashMap = new HashMap<>();

        if (s.length() != t.length()) {
            return false;
        }

        for (char ch : t.toCharArray()) {
            if (hashMap.containsKey(ch)) {
                hashMap.put(ch, hashMap.get(ch) + 1);
            } else {
                hashMap.put(ch, 1);
            }
        }

        for (char ch : s.toCharArray()) {
            if (hashMap.containsKey(ch)) {
                hashMap.put(ch, hashMap.get(ch) - 1);
            } else {
                return false;
            }
        }

        for (char ch : hashMap.keySet()) {
            if (hashMap.get(ch) != 0) {
                return false;
            }
        }

        return true;
    }
}